src/autogen.ml

```diff
@@ -9,24 +9,15 @@ let mk_mapper file =
 let mk_output dir file =
   Filename.concat dir (file ^ ".ml")
 
-let make_file_nicer file =
+(* template.ml is what the student is going to write into. Therefore, we want
+ * to modify a bit the output of the pretty-printer to add a white line between
+ * each function of the exercise. We also remove trailing whitespaces. *)
+let add_linebreaks file =
   let _ = Sys.command ("sed -Ei 's/[ \\t]+$//' " ^ file) in
   let basename = Filename.basename file in
-(*
-  let with_mutually_rec_functions =
-    Sys.command ("grep '^and\\s' " ^ file ^ " | wc -l && exit $?") > 0 in
-*)
-  let add_spacing = (
-    if basename = "template.ml" || basename = "solution.ml"
-    (*&& not with_mutually_rec_functions *)
-    then
-(*       "s/(= \".*\")/\\1 ;;/; /;;/G" *)
-       "/$/G"
-    else
-      "")
- in
-(*   Printf.printf "sed -Ei '%s' %s" add_spacing file *)
-  ignore (Sys.command (Printf.sprintf "sed -Ei '%s' %s" add_spacing file))
+  if basename = "template.ml" || basename = "solution.ml" then
+    ignore (Sys.command ("sed -Ei '/$/G' " ^ file))
```

Comment:
Ignoring exit status is dangerous: please introduce a cleaner error handling mechanism.